frame/example-offchain-worker/src/lib.rs

```diff
@@ -197,7 +197,12 @@ decl_storage! {
 
 impl<T: Trait> Module<T> {
     fn offchain_worker_main(block_number: T::BlockNumber) -> ResultStr<()> {
-        let signer = Self::get_signer()?;
+        let signer = Self::get_signer();
+
+		if let Err(e) = signer {
```

Comment:
FYI the canonical way of doing these things is using `match` and avoid `unwrap`.

    let signer = match Self::get_signer() {
        Err(e) => { … }
        Ok(signer) => signer,
    };

Your style reminds me of golang which is also good though.